conans/model/scm.py

```diff
@@ -1,34 +1,51 @@
-import json
 import os
 import subprocess
 
+from six import string_types
+
 from conans.client.tools.scm import Git, SVN
 from conans.errors import ConanException
 from conans.util.files import rmdir
 
 
 def get_scm_data(conanfile):
-    try:
+    data = getattr(conanfile, "scm", None)
+    if data is not None and isinstance(data, dict):
         return SCMData(conanfile)
-    except ConanException:
+    else:
         return None
 
 
+def _get_dict_value(data, key, expected_type, default=None, disallowed_type=None):
+    if key in data:
+        r = data.get(key)
+        if r is None:  # None is always a valid value
+            return r
```

Comment:
`None` has to be a valid value right now. There are many recipes using functions like the following one to get SCM values:

```python
def get_revision():
    try:
        return Git(...).get_commit()
    except:
        return None

```